drakrun/drakrun/regression.py

```diff
@@ -0,0 +1,140 @@
+import argparse
+import hashlib
+import json
+import os
+import threading
+import tempfile
+import logging
+from dataclasses import dataclass
+from typing import Optional, List
+from pathlib import Path
+
+from karton.core import Karton, Task, Producer, Resource, Config
+from malduck.extractor import ExtractManager, ExtractorModules
+from mwdblib import MWDB
+from drakrun.version import __version__ as DRAKRUN_VERSION
+from contextlib import contextmanager
+
+
+@contextmanager
+def changedLogLevel(logger, level):
+    old_level = logger.level
+    logger.setLevel(level)
+    yield
+    logger.setLevel(old_level)
+
+
+@dataclass
+class TestCase:
+    sha256: str
+    extension: str
+    ripped: str
+    path: Optional[str] = None
+
+    def get_sample(self) -> bytes:
+        # If user provided file path for the sample
+        if self.path:
+            data = open(self.path, "rb").read()
+
+            h = hashlib.sha256()
+            h.update(data)
+            if h.hexdigest() != self.sha256:
+                raise RuntimeError(f"Expected {self.sha256}, got {h.hexdigest()}")
+            return data
+
+        # Otherwise try to fetch the sample from MWDB
+        mwdb = MWDB(api_key=os.getenv("MWDB_API_KEY"))
+        sample = mwdb.query_file(self.sha256)
+        return sample.download()
```

Comment:
maybe some cache? like `mwdb/<sha256>` as a filename to avoid downloading same samples over and over

Reply:
I've thought about this and came to the conclusion that if one wants to have local samples, they can use the `path` value.
Additionally, I've asked about caching on the mwdblib level and it was rejected.

All in all, this code won't be executed very frequently and thus fetching samples from MWDB shouldn't be an issue.

Reply:
ok